File: UI_Flutter/minifootballfront/lib/user_filter.dart
class UserFilter {
  int? id;
  int? countryId;
  int? cityId;
  bool? isBanned;
  String? name;
  String? email;
  int? dayOfBirth;
  int? monthOfBirth;
  int? yearOfBirth;
  bool? isAdmin;
  bool? orderByAgeAsc;
  bool? orderByCreationDateAsc;

  UserFilter({
    this.id,
    this.countryId,
    this.cityId,
    this.isBanned,
    this.name,
    this.email,
    this.dayOfBirth,
    this.monthOfBirth,
    this.yearOfBirth,
    this.isAdmin,
    this.orderByAgeAsc,
    this.orderByCreationDateAsc,
  });


  factory UserFilter.fromJson(Map<String, dynamic> json) {
    return UserFilter(
      id: json['id'],
      countryId: json['countryId'],
      cityId: json['cityId'],
      isBanned: json['isBanned'],
      name: json['name'],
      email: json['email'],
      dayOfBirth: json['dayOfBirth'],
      monthOfBirth: json['monthOfBirth'],
      yearOfBirth: json['yearOfBirth'],
      isAdmin: json['isAdmin'],
      orderByAgeAsc: json['orderByAgeAsc'],
      orderByCreationDateAsc: json['orderByCreationDateAsc'],
    );
  }


  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'countryId': countryId,
      'cityId': cityId,
      'isBanned': isBanned,
      'name': name,
      'email': email,
      'dayOfBirth': dayOfBirth,
      'monthOfBirth': monthOfBirth,
      'yearOfBirth': yearOfBirth,
      'isAdmin': isAdmin,
      'orderByAgeAsc': orderByAgeAsc,
      'orderByCreationDateAsc': orderByCreationDateAsc,
    };
  }
}

File: MinifootballAppApi/MinifootballAppApi/DTOs/Filters/UserFilter.cs
namespace MinifootballAppApi.DTOs.Filters
{
    public class UserFilter
    {
        public int? Id { get; set; }
        public int? CountryId { get; set; }
        public int? CityId { get; set; }
        public bool? IsBanned { get; set; }
        public string? Name { get; set; }
        public string? Email { get; set; }
        public int? DayOfBirth { get; set; }
        public int? MonthOfBirth { get; set; }
        public int? YearOfBirth { get; set; }
        public bool? IsAdmin { get; set; }
        public bool? OrderByAgeAsc { get; set; }
        public bool? OrderByCreationDateAsc { get; set; }

    }
}
